[Bug]: wolfSSL_connect returns bad error code

### Contact Details

tk@nabto.com

### Version

master

### Description

ssl.c:11982 stores the error returned by `SendBuffered()` directly in `ssl->error`
ssl.c:12000 it is assumed to be stored in `ret`
this results in `wolfSSL_connect()` returning `SSL_FATAL_ERROR` with `wolfSSL_get_error()` returning `OK`.
When using non-blocking IO, this causes a harmless `WANT_WRITE` into a fatal error.

possible fix: Delete ssl.c:12000

```
diff --git a/src/ssl.c b/src/ssl.c
index ead087c95..b14349936 100644
--- a/src/ssl.c
+++ b/src/ssl.c
@@ -11980,41 +11980,40 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
         #endif
         ) {
             if ( (ssl->error = SendBuffered(ssl)) == 0) {
                 if (ssl->fragOffset == 0 && !ssl->options.buildingMsg) {
                     if (advanceState) {
                         ssl->options.connectState++;
                         WOLFSSL_MSG("connect state: "
                                     "Advanced from last buffered fragment send");
                     }
                 #ifdef WOLFSSL_ASYNC_IO
                     /* Cleanup async */
                     FreeAsyncCtx(ssl, 0);
                 #endif
                 }
                 else {
                     WOLFSSL_MSG("connect state: "
                                 "Not advanced, more fragments to send");
                 }
             }
             else {
-                ssl->error = ret;
                 WOLFSSL_ERROR(ssl->error);
                 return WOLFSSL_FATAL_ERROR;
             }
         }

```

### Reproduction steps

_No response_

### Relevant log output

```shell
2022-07-06 08:29:44.314 wolfssl(000)[TRACE] wolfSSL Entering BuildMessage
2022-07-06 08:29:44.315 wolfssl(000)[TRACE] wolfSSL Leaving BuildMessage, return 0
2022-07-06 08:29:44.315 wolfssl(000)[TRACE] wolfSSL Leaving DtlsMsgPoolSend(), return -327
2022-07-06 08:29:44.316 wolfssl(000)[ERROR] wolfSSL error occurred, error = -327
2022-07-06 08:29:44.316 wolfssl(000)[TRACE] wolfSSL Leaving wolfSSL_dtls_got_timeout(), return -1
2022-07-06 08:29:44.316 wolfssl(000)[TRACE] wolfSSL Entering SSL_get_error
2022-07-06 08:29:44.316 wolfssl(000)[TRACE] wolfSSL Leaving SSL_get_error, return -327
2022-07-06 08:29:44.317 ...wolfssl/nm_wolfssl_cli.c(531)[TRACE] Got timeout returned want read/write
2022-07-06 08:29:44.317 wolfssl(000)[TRACE] wolfSSL Leaving wolfSSL_dtls_get_current_timeout(), return 2
2022-07-06 08:29:44.318 ...bevent/nm_libevent_udp.c(343)[TRACE] Received udp packet of size 1215, from 0000:0000:0000:0000:0000:ffff:7f00:0001, port 58645 
2022-07-06 08:29:44.318 wolfssl(000)[TRACE] wolfSSL Entering SSL_connect()
2022-07-06 08:29:44.319 wolfssl(000)[ERROR] wolfSSL error occurred, error = 0 line:11968 file:src/ssl.c
2022-07-06 08:29:44.320 wolfssl(000)[TRACE] wolfSSL Entering SSL_get_error
2022-07-06 08:29:44.320 wolfssl(000)[TRACE] wolfSSL Leaving SSL_get_error, return 0
2022-07-06 08:29:44.320 wolfssl(000)[TRACE] wolfSSL Entering ERR_error_string
2022-07-06 08:29:44.322 ...wolfssl/nm_wolfssl_cli.c(442)[ERROR] Certificate verification failed: (0) ok
2022-07-06 08:29:44.322 ...wolfssl/nm_wolfssl_cli.c(456)[_INFO] wolfssl_connect returned -1, which is 0, ok
```

--- a/src/dtls13.c
+++ b/src/dtls13.c
@@ -1969,7 +1969,7 @@ int Dtls13SetRecordNumberKeys(WOLFSSL* ssl, enum encrypt_side side)
 {
     RecordNumberCiphers* enc = NULL;
     RecordNumberCiphers* dec = NULL;
-    byte *encKey, *decKey;
+    byte *encKey = NULL, *decKey = NULL;
     int ret;
 
     if (ssl == NULL) {

--- a/src/ssl.c
+++ b/src/ssl.c
@@ -11983,7 +11983,8 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
             && ssl->error != WC_PENDING_E
         #endif
         ) {
-            if ( (ssl->error = SendBuffered(ssl)) == 0) {
+            ret = SendBuffered(ssl);
+            if (ret == 0) {
                 if (ssl->fragOffset == 0 && !ssl->options.buildingMsg) {
                     if (advanceState) {
                         ssl->options.connectState++;
@@ -12501,7 +12502,8 @@ int wolfSSL_DTLS_SetCookieSecret(WOLFSSL* ssl,
             && ssl->error != WC_PENDING_E
         #endif
         ) {
-            if ( (ret = SendBuffered(ssl)) == 0) {
+            ret = SendBuffered(ssl);
+            if (ret == 0) {
                 /* fragOffset is non-zero when sending fragments. On the last
                  * fragment, fragOffset is zero again, and the state can be
                  * advanced. */

